### PR TITLE
Refactor application workflow closeChannel

### PR DESCRIPTION
**Pull request by snario**
Begins to address #1967

### CODE REVIEW

File: packages/xstate-wallet/src/workflows/application.ts [hunk 1]
```diff
@@ -350,13 +331,39 @@ export const workflow = (
...
+        } catch (error) {
+          const matches = reason => new RegExp(reason).test(error.message);
+
+          // TODO: Catch other errors
+          let message: ErrorResponse['error'];
+          if (matches(Errors.notMyTurn)) message = {code: 403, message: 'Not your turn'};
+          else if (matches(Errors.channelMissing))
+            message = {code: 400, message: 'Channel not found'};
+          else {
+            message = {code: 500, message: 'Wallet error'};
```
Comment (andrewgordstewart) [inline packages/xstate-wallet/src/workflows/application.ts:363]: So I arbitrarily chose to _not_ repeat error codes for different API calls. I don't know if that's the right choice, but we should either stick with it, and use the `CloseChannel` error codes, or change that choice.

https://github.com/statechannels/monorepo/blob/master/packages/client-api-schema/src/error-codes.ts#L8-L11

File: packages/xstate-wallet/src/workflows/tests/application.test.ts [hunk 2]
```diff
@@ -197,8 +197,8 @@ it('starts concluding when requested', async () => {
   service.start('running');
   service.send({type: 'PLAYER_REQUEST_CONCLUDE', channelId});
   await waitForExpect(async () => {
-    expect(service.state.value).toEqual('attemptToSignFinalState');
-    expect(services.signFinalStateIfMyTurn).toHaveBeenCalled();
+    expect(service.state.value).toEqual('running');
+    expect(actions.closeChannel).toHaveBeenCalled();
   }, 2000);
 });
```
[the remark this answers is not in the document]
Comment (snario) [reply packages/xstate-wallet/src/workflows/tests/application.test.ts:203]: Do you mean `running.test.ts`?

[the remark this answers is not in the document]
Comment (andrewgordstewart) [reply packages/xstate-wallet/src/workflows/tests/application.test.ts:203]: That's the one

[the remark this answers is not in the document]
Comment (andrewgordstewart) [reply packages/xstate-wallet/src/workflows/tests/application.test.ts:203]: https://github.com/statechannels/monorepo/blob/master/packages/xstate-wallet/src/integration-tests/running.test.ts#L89-L103

(Makes sense to do it as a separate PR, too)

**Pull request merged**